### ISSUE TITLE
Feature Request - gitlab_deploy_key_enable with "can_push" argument

### ISSUE THREAD

**Issue by guillemtp** [gitlabhq/terraform-provider-gitlab#577]
### Feature Request Details

Hi! Please add the ability to configure the argument "can_push" in the "gitlab_deploy_key_enable" resource. This will provide the ability to enable an existent key in a repository with write permissions. Otherwise, when you enable an existing deploy_key in a repository is done with only read permissions. 

Thanks

**Comment by armsnyder** [comment]
This attribute exists already (`can_push`) but it isn't documented unfortunately.

**Comment by guillemtp** [comment]
In my scenario, the `gitlab_deploy_key` is managed also with Terraform with the `can_push`setted to `true`. And this works where is created.

My code is the following. The deploy key creation in specific project:

```
resource "gitlab_deploy_key" "gitlab_ci" {
  project = gitlab_project.ci_templates.path_with_namespace
  title   = "CI Deploy key - Terraform managed"
  key     = aws_ssm_parameter.gitlab_deploy_key_public.value
  can_push = true
}
```

And enabling for each other project:
```
resource "gitlab_deploy_key_enable" "gitlab_ci" {
  project = gitlab_project.main.id
  key_id = var.deploy_key_id
  can_push = true
}
```

But on the projects where's is enabled, it doesn't work. It's true that `can_push` attribute exists on `gitlab_deploy_key_enable` but just now I've tried to add and it doesn't takes effect. The deploy key is enabled in the project but without the write permissions. 

I think the problem is the gitlab api doesn't have the `can_push`parameter on the enable call https://docs.gitlab.com/ce/api/deploy_keys.html#enable-a-deploy-key and we can only enable it making a second call to the update api endpoint for the project.

My workaround to get it is to add this call in Terraform, like this:

```
resource "null_resource" "api_deploy_key_write" {
  triggers = {
    version = 1
  }
  provisioner "local-exec" {
    command = "curl --request PUT --header \"PRIVATE-TOKEN: $GITLAB_TOKEN\" --header \"Content-Type: application/json\"  --data '{\"can_push\": true}' \"https://gitlab.com/api/v4/projects/$PROJECT_ID/deploy_keys/$DEPLOY_KEY_ID\""
    environment = {
      PROJECT_ID = gitlab_project.main.id
      GITLAB_TOKEN = var.gitlab_token
      DEPLOY_KEY_ID = var.deploy_key_id
    }
  }
  depends_on = [
    gitlab_deploy_key_enable.gitlab_ci
  ]
}
```

Take note that this update is not in the project where de Deploy Key is created, but in the projecte where the deploy key is enabled.

**Comment by guillemtp** [comment]
Maybe, after this line https://github.com/gitlabhq/terraform-provider-gitlab/blob/34ef2f9926fddea87efc8b8bc31e1beb1cb93964/gitlab/resource_gitlab_deploy_key_enable.go#L59 should be called `client.DeployKeys.UpdateDeployKey`function?

**Comment by armsnyder** [comment]
There's a PR open for this - #608

**Comment by timofurrer** [comment]
Implemented in #1009 and will be release with the next release 🎉
